src/features/pages/PlaySong/index.tsx

```diff
@@ -33,10 +35,17 @@ export function PlaySong() {
   const player = Player.player()
   const synth = useSingleton(() => getSynthStub('acoustic_grand_piano'))
   let { data: song, error } = useSong(id, source)
+  let songMeta = useSongMetadata(id, source) 
+
+  if (source === 'base64') {
```

Comment:
This logic should be part of `useSong`. It already will have differing behavior depending on the `source` -- that way other parts of the app wouldn't need to duplicate this if condition.

Reply:
done